package.json

```diff
@@ -24,7 +24,7 @@
   "homepage": "https://github.com/Portchain/logacious#readme",
   "dependencies": {
     "ain2": "2.0.0",
-    "lodash": "4.17.11",
+    "lodash": "^4.17.20",
```

Comment:
Is your preference to use version ranges and __package-lock.json__ rather than specifying exact versions in __package.json__? I realize they are not quite equivalent as specifying exact versions in __package.json__ does not prevent transient dependencies from changing.

I think of __package-lock.json__ as belonging at the application level rather than the library level, but I suppose it does not hurt to use lock files at both levels. :)

Reply:
@davidchambers I'm committing `package-lock.json` because `npm audit` requires the `package-lock.json` file to work. 

We don't have CI set up for this repository yet, but if we set it up in the future, we'll need the `package-lock.json` to check security vulnerabilities via `npm audit` as part of CI.

Reply:
I did not know about this requirement. :thumbsup:

Reply:
@utkuufuk package-lock.json belongs at the application level, not at the library level. Consider an application `A` that uses libraries `B1`, `B2`, `B3` that each use a common dependency `C` (such as lodash). If each of those libraries used a package lock, then the top level application could potentially have 3 different versions of dependency `C`, which bloats npm_modules and the final bundle sent to the frontend. Instead, libraries should use version ranges.

You are absolutely right that we should use `npm audit` and other tools to check for vulnerabilities. However, I think it is sufficient to run the audit at the application level (because it also looks at transitive dependencies).

Reply:
@SergeiPatiakin That makes sense, thanks for pointing it out. I'll open another PR to remove `package-lock.json`. Good thing that we haven't published the new version to NPM yet.